datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -302,6 +287,8 @@ impl ExecutionPlan for ParquetExec {
                     })
             })?;
 
+        let config_options = ctx.session_config().config_options();
```

Comment:
We need to fetch this at execution time, in order that datafusion-proto can still deserialize `ParquetExec` without a `SessionState`. Longer term as we strip out the overrides this will make more sense anyway so :shrug:

Reply:
I think it is reasonable to look at the session configuration while executing 🤷  

It certainly seems better than the current state of master where the config options (attached to session state) are read via interior mutability